project/glossary.md

```diff
@@ -0,0 +1,46 @@
+<!-- vim: ft=markdown colorcolumn=72
+-->
+# Glossary
+
+This glossary is a dictionary of terms defined as they are used by the
+spire/helm-charts project.  Writing the definitions of terms here helps
+to keep other documents concise and precise.  Documenting terminology
+here helps prevent misundersandings, facilitating easy onboarding of new
+team members.
+
+**Deployment**
+: A use of the Helm Charts to describe a single SPIRE cluster.
```

Comment:
Since there are multiple "clusters" involved, should we call it out explicitly like done here? Throughout the rest of the document cluster is being used to talk about a SPIRE cluster and not a k8s cluster or sql cluster.

Or document Cluster as its own term? That may be better too, as then we can be explicit about what more then a multi pod spire-server makes up a "spire cluster"?

Reply:
Yes, we should be specific where it helps, and non-specific where it hurts.

I wouldn't want a definition of "Cluster" to be more than "A collection of related computers and software acting as part of a system"  

The alternative is "Cluster" means "A collection of related computers and software acting as part of SPIRE deployment" which would then require Kubernetes Cluster to mean one of "A Kubernetes collection of related computers and software acting as part of a SPIRE deployment" (blocking the ability to talk about Kubernetes clusters) or "A collection of related computers and software acting as part of a Kubernetes deployment" (making contextual comments of "Kubernetes Cluster" with follow on "Cluster" comments contextually awkward.

Feel free to suggest "SPIRE Cluster" / "Kubernetes Cluster" where you see a raw "Cluster" and the meaning should be constrained to only one or the other.